website/docs/TestingEnvironment.md

```diff
@@ -0,0 +1,160 @@
+
+```jsx
+* <Pressable> (composite)
+  * <View accessible={true} {...}> (host)
+    * children prop passed in JSX
+```
+
+#### Differentiating between host and composite elements
+
+Any easy way to differentiate between host and composite elements is the `type` prop of `ReactTestInstance`:
+
+- for host components it’s always a string value representing component name, e.g. `"View"`
+- for composite components it’s function or class defining the component
+
+You can use the following code to check if given element is a host one:
+
+```jsx
+function isHostElement(element: ReactTestInstance) {
+  return typeof element.type === 'string';
+}
+```
+
+### Tree nodes
+
+We encourage you to only assert values on host views in your tests, because they represent the user interface view and controls that the user will be able to see and interact with. Users cannot see, or interact with, composite views as they exist purely in JavaScript domain and do not generate any visible UI.
+
+#### Asserting props
+
+As an example, if you make assertions on a `style` prop of a composite element, there is no guarantee that the style will be visible to the user, as the component author can forget to pass this prop to some underlying `View` or other host component. Similarly `onPress` event handler on a composite prop can be unreachable by the user.
+
+```jsx
+function ForgotToPassPropsButton({ title, onPress, style }) {
+  return (
+    <Pressable>
+      <Text>{title}</Text>
+    </Pressable>
+  );
+}
+```
+
+In the above example user defined components accepts both `onPress` and `style` props but does not pass it (through `Pressable`) to host views, so these props will not affect the user interface.
+
+### Tree navigation
+
+When navigating three using `parent` or `children` props of `ReactTestInstance` element, you will encounter both host and composite elements. You should be careful when navigating the element tree, as the tree structure for 3rd party components and change independently from your code and cause unexpected test failures.
+
+If you want to find a host element for given element they you might use following code:
+
+```jsx
+function getHostParent(element: ReactTestInstance) {
+  let current = element.parent;
+  while (current) {
+    if (isHostElement(current)) {
+      return current;
+    }
+
+    current = current.parent;
+  }
+
+  return null;
+}
+```
+
+### Queries
+
+Most of the Testing Library queries return host components, in order to encourage best practices described above.
+
+At this stage, there are some noteworthy exceptions:
+
+- `*ByText` queries returns composite `Text` element
+- `*ByDisplayValue` queries returns composite `TextInput` element
+- `*ByPlaceholderText` queries returns composite `TextInput` element
+
+This will change in the near future, as we make efforts for all queries to return host components. Meanwhile it should be a huge issue, as composite `Text` and `TextInput`generally pass their props down to host counterparts.
```

Comment:
```suggestion
This will change in the near future, as we make efforts for all queries to return host components. Meanwhile it shouldn't be a huge issue, as composite `Text` and `TextInput`generally pass their props down to host counterparts.
```